drivers/spi/spi_simple_spi.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2019 Western Digital Corporation or its affiliates
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(spi_simple_spi);
+
+#include <sys_io.h>
+#include <spi.h>
+
+#include "spi_context.h"
+#include "spi_simple_spi.h"
+
+/* Bit 5:4 == ESPR, Bit 1:0 == SPR */
+u8_t DIVIDERS[] = { 0x00,       /*   2  */
+		    0x01,       /*   4  */
+		    0x10,       /*   8  */
+		    0x02,       /*  16  */
+		    0x03,       /*  32  */
+		    0x11,       /*  64  */
+		    0x12,       /* 128  */
+		    0x13,       /* 256  */
+		    0x20,       /* 512  */
+		    0x21,       /* 1024 */
+		    0x22,       /* 2048 */
+		    0x23 };     /* 4096 */
+
+int simple_spi_transceive(struct device *dev,
+			  const struct spi_config *config,
+			  const struct spi_buf_set *tx_bufs,
+			  const struct spi_buf_set *rx_bufs)
+{
+	struct spi_context *ctx = &SPI_DATA(dev)->ctx;
+
+	u8_t rx_byte;
+
+	/* Lock the SPI Context */
+	spi_context_lock(ctx, false, NULL);
+
+	/* Set clock divider */
+	int i;
+
+	for (i = 0; i < 12; i++)
+		if ((config->frequency << (i + 1)) >
+		    CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC) {
+			break;
+		}
+
+	sys_write8((DIVIDERS[i] >> 4) & 0x3, SPI_REG(dev, SPI_SPER));
+	u8_t spcr = DIVIDERS[i] & 0x3;
+
+	/* Set chip select */
+	if (config->cs == NULL) {
+		sys_write8(1 << config->slave, SPI_REG(dev, SPI_SPSS));
+	}
```

Comment:
please, support as well the possibility to emulate the cs line via a gpio line. See other drivers, calling spi_context_cs_configure() (then spi_context_cs_control() )

Reply:
Done